components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSPortalAppMigrationTool.java

```diff
@@ -0,0 +1,258 @@
+                        migrateLayoutsInStore(artifactTypes[artifactCount]);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * migrate layouts into newer version
+     *
+     * @param layouts file path of the layouts directory
+     */
+    private void migrateLayoutsInStore(File layouts) {
+        File[] listOflayouts = layouts.listFiles();
+        JSONParser parser = new JSONParser();
+        for (int layoutCount = 0; layoutCount < listOflayouts.length; layoutCount++) {
+            try {
+                JSONObject layoutObj = (JSONObject) parser.parse(new FileReader(
+                        listOflayouts[layoutCount].getAbsolutePath() + File.separator + INDEX_JSON));
+                updateDashboardBlocks(((JSONArray) layoutObj.get(BLOCKS)));
+                FileWriter file = new FileWriter(
+                        listOflayouts[layoutCount].getAbsolutePath() + File.separator + INDEX_JSON);
+                file.write(layoutObj.toJSONString());
+                file.flush();
+                file.close();
+
+            } catch (IOException e) {
+                log.error("Error in opening the file " + listOflayouts[layoutCount].getName(), e);
+            } catch (ParseException e) {
+                log.error("Error in parsing the index.json file in " + listOflayouts[layoutCount].getAbsolutePath());
+            }
+        }
+    }
+
+    private TrustManager[] get_trust_mgr() {
+        TrustManager[] certs = new TrustManager[] { new X509TrustManager() {
+            public X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+
+            public void checkClientTrusted(X509Certificate[] certs, String t) {
+            }
+
+            public void checkServerTrusted(X509Certificate[] certs, String t) {
+            }
+        } };
+        return certs;
+    }
+
+    public void updateDashboardJSON() {
+        String response = invokeRestAPI("https://localhost:9443/portal/apis/login?username=admin&password=admin",
+                "POST", null, null);
+        try {
+            JSONObject responseObj = (JSONObject) new JSONParser().parse(response.toString());
+            String sessionId = (String) responseObj.get("sessionId");
+            response = invokeRestAPI("https://localhost:9443/portal/apis/dashboards", "GET", sessionId, null);
+            JSONArray responseArr = (JSONArray) new JSONParser().parse(response.toString());
+            for (int dashboardCount = 0; dashboardCount < responseArr.size(); dashboardCount++) {
+                modifyDashboardJSON(dashboardUpdater((JSONObject) responseArr.get(dashboardCount)), sessionId);
+            }
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public String invokeRestAPI(String requestURL, String requestType, String sessionId, String content) {
+        try {
+            System.setProperty("javax.net.ssl.trustStore",
```

Comment:
Read the trustStore location, etc from the configuration. If not provided load it from default location.